From 1dc00b5a70cda80b0efebefe33fbdcec982096b4 Mon Sep 17 00:00:00 2001
From: eloimuns <eloimuns@gmail.com>
Date: Tue, 12 Nov 2013 21:05:58 +0100
Subject: [PATCH] PAC patch: TRDS 2.0

---
 include/android/configuration.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/include/android/configuration.h b/include/android/configuration.h
index d5cddb3..1b5bfa5 100644
--- a/include/android/configuration.h
+++ b/include/android/configuration.h
@@ -77,6 +77,11 @@ enum {
     ACONFIGURATION_SCREENLONG_NO = 0x1,
     ACONFIGURATION_SCREENLONG_YES = 0x2,
 
+    ACONFIGURATION_UI_INVERTED_MODE_ANY = 0x0000,
+    ACONFIGURATION_UI_INVERTED_MODE_NORMAL = 0x0001,
+    ACONFIGURATION_UI_INVERTED_MODE_YES = 0x0002,
+    ACONFIGURATION_UI_INVERTED_MODE_NO = 0x0003,
+
     ACONFIGURATION_UI_MODE_TYPE_ANY = 0x00,
     ACONFIGURATION_UI_MODE_TYPE_NORMAL = 0x01,
     ACONFIGURATION_UI_MODE_TYPE_DESK = 0x02,
@@ -110,6 +115,7 @@ enum {
     ACONFIGURATION_SCREEN_SIZE = 0x0200,
     ACONFIGURATION_VERSION = 0x0400,
     ACONFIGURATION_SCREEN_LAYOUT = 0x0800,
+    ACONFIGURATION_UI_INVERTED_MODE = 0x8000,
     ACONFIGURATION_UI_MODE = 0x1000,
     ACONFIGURATION_SMALLEST_SCREEN_SIZE = 0x2000,
     ACONFIGURATION_LAYOUTDIR = 0x4000,
@@ -284,6 +290,16 @@ int32_t AConfiguration_getScreenLong(AConfiguration* config);
 void AConfiguration_setScreenLong(AConfiguration* config, int32_t screenLong);
 
 /**
+ * Return the current ACONFIGURATION_UI_INVERTED_MODE_* set in the configuration.
+ */
+int32_t AConfiguration_getUiInvertedMode(AConfiguration* config);
+
+/**
+ * Set the current inverted UI mode in the configuration.
+ */
+void AConfiguration_setUiInvertedMode(AConfiguration* config, int32_t uiInvertedMode);
+
+/**
  * Return the current ACONFIGURATION_UI_MODE_TYPE_* set in the configuration.
  */
 int32_t AConfiguration_getUiModeType(AConfiguration* config);
-- 
1.8.1.2

